test-d/types/helperTypes.test-d.ts

```diff
@@ -0,0 +1,63 @@
+import { expectType } from "tsd";
+import type { ArrayOrTupleKey, DotNotationKeys, GetValueFromDotKey } from "../../src/types/helperTypes";
+
+declare function type<T>(): T;
+
+// basic objects
+expectType<"k1">(type<DotNotationKeys<{ k1: 1 }>>());
```

Comment:
While it's not a bad idea to start testing our types I think the `type` function helper here is clumsy.
Try this:
```ts
import { TypeEqual } from "ts-expect";

expectType<TypeEqual<DotNotationKeys{ k1: 1 }>, "k1">(true);
```
It's still a a bit awkward but it's at least more expressive and something shown in `tsd`'s examples.